src/ui/api/repository.go

```diff
@@ -319,41 +321,115 @@ func (ra *RepositoryAPI) GetTags() {
 		log.Errorf("error occurred while initializing repository client for %s: %v", repoName, err)
 		ra.CustomAbort(http.StatusInternalServerError, "internal error")
 	}
-	tags, err := listTag(client)
+
+	// get tags
+	tags, err := getDetailedTags(client)
 	if err != nil {
 		regErr, ok := err.(*registry_error.Error)
 		if !ok {
-			log.Errorf("error occurred while listing tags of %s: %v", repoName, err)
-			ra.CustomAbort(http.StatusInternalServerError, "internal error")
+			ra.HandleInternalServerError(fmt.Sprintf(
+				"failed to list tags of repository %s: %v", repoName, err))
+			return
 		}
-
-		ra.CustomAbort(regErr.StatusCode, regErr.Detail)
+		ra.RenderError(regErr.StatusCode, regErr.Detail)
+		return
 	}
 
-	result := []tagResp{}
+	// get signatures
+	signatures := map[string]*notary.Target{}
+	if config.WithNotary() {
+		signatures, err = getSignatures(repoName, ra.SecurityCtx.GetUsername())
+		if err != nil {
+			ra.HandleInternalServerError(fmt.Sprintf(
+				"failed to get signatures of repository %s: %v", repoName, err))
+			return
+		}
+	}
 
+	// assemble the response
+	tagResps := []*tagResp{}
 	for _, tag := range tags {
-		manifest, err := getManifest(client, tag, "v1")
-		if err != nil {
-			if regErr, ok := err.(*registry_error.Error); ok {
-				ra.CustomAbort(regErr.StatusCode, regErr.Detail)
-			}
+		tagResp := &tagResp{
+			tag: *tag,
+		}
 
-			log.Errorf("failed to get manifest of %s:%s: %v", repoName, tag, err)
-			ra.CustomAbort(http.StatusInternalServerError, "internal error")
+		// compare both digest and tag
+		if signature, ok := signatures[tag.Digest]; ok {
+			if tag.Name == signature.Tag {
+				tagResp.Signature = signature
+			}
 		}
 
-		result = append(result, tagResp{
-			Tag:      tag,
-			Manifest: manifest.Manifest,
-		})
+		tagResps = append(tagResps, tagResp)
 	}
 
-	ra.Data["json"] = result
+	ra.Data["json"] = tagResps
 	ra.ServeJSON()
 }
 
-func listTag(client *registry.Repository) ([]string, error) {
+// get tags of the repository, read manifest for every tag
+// and assemble necessary attrs(os, architecture, etc.) into
+// one struct
+func getDetailedTags(client *registry.Repository) ([]*tag, error) {
+	ts, err := getSimpleTags(client)
+	if err != nil {
+		return nil, err
+	}
+
+	list := []*tag{}
+
+	for _, t := range ts {
+		digest, manifest, config, err := getV2Manifest(client, t)
+		if err != nil {
+			return nil, err
+		}
+
+		// manifest can be used to calculate the image size
+		_ = manifest
```

Comment:
move the comment to line 382, and ignore the manifest in line 382 
"digest, _, config, err := xxxxxxxx"